ironfish/src/blockchain/blockchain.test.ts

```diff
@@ -759,74 +750,44 @@ describe('Blockchain', () => {
     // Now create the double spend
     await node.wallet.updateHead()
     const tx = await useTxFixture(node.wallet, accountA, accountB)
-    const doubleSpend = tx.getSpend(0)
-
-    const treeSize = await node.chain.nullifiers.size()
-
-    // The nullifier is not found in the tree
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize),
-    ).resolves.toBe(false)
 
-    // Let's spend the transaaction for the first time
+    // Spend the transaaction for the first time
     const block3 = await useMinerBlockFixture(node.chain, 3, undefined, undefined, [tx])
     await expect(node.chain).toAddBlock(block3)
 
-    // The nullifier is not found at the old tree size, but is found at the new tree size
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(true)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize),
-    ).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize + tx.spendsLength()),
-    ).resolves.toBe(true)
-
-    // Let's spend the transaction a second time
+    // Spend the transaction a second time
     const block4 = await useMinerBlockFixture(node.chain, 4, undefined, undefined, [tx])
-    await expect(node.chain).toAddBlock(block4)
-
-    // We've now added a double spend, we can see that because of a bug in the
-    // MerkleTree implementation the nullifier is no longer found at the tree
-    // size before this block, but moved to the tree size at block3
-    await expect(node.chain.nullifiers.contains(doubleSpend.nullifier)).resolves.toBe(true)
-    await expect(
-      node.chain.nullifiers.contained(doubleSpend.nullifier, treeSize + tx.spendsLength()),
-    ).resolves.toBe(false)
-    await expect(
-      node.chain.nullifiers.contained(
-        doubleSpend.nullifier,
-        treeSize + tx.spendsLength() + tx.spendsLength(),
-      ),
-    ).resolves.toBe(true)
-
-    // Now we set our fix to activate at sequence 5 so this block will not let
-    // us add the transaction a third time
-    const block5 = await useMinerBlockFixture(node.chain, 5, undefined, undefined, [tx])
-    await expect(node.chain.addBlock(block5)).resolves.toMatchObject({
+    await expect(node.chain.addBlock(block4)).resolves.toMatchObject({
       isAdded: false,
       reason: VerificationResultReason.DOUBLE_SPEND,
     })
   })
 
   it('rejects double spend during reorg', async () => {
-    // G -> A2 -> A3
-    //   -> B2 -> B3* -> A4
+    /**
+     * We do check double spends when connecting forks because we don't rebuild the nullifier
```

Comment:
This sentence is inverted I think.

```suggestion
     * We don't check double spends when connecting forks because we don't rebuild the nullifier
```